plutus-ledger-constraints/src/Ledger/Constraints/OnChain/V2.hs

```diff
@@ -108,7 +113,7 @@ checkTxConstraint ctx@ScriptContext{scriptContextTxInfo} = \case
         $ maybe False (isNoOutputDatum . txOutDatum . txInInfoResolved) (PV2.findTxInByTxOutRef txOutRef scriptContextTxInfo)
     MustSpendScriptOutput txOutRef rdmr ->
         traceIfFalse "L8" -- "Script output not spent"
-        $ rdmr `elem` (txInfoRedeemers scriptContextTxInfo)
+        $ rdmr == fromJust' "L8" (AMap.lookup (Spending txOutRef) (txInfoRedeemers scriptContextTxInfo))
```

Comment:
How useful is this?

This is using as existing code (L8), so maybe have a different one?

Reply:
> How useful is this?

Without this change it could pass if some other script's redeemer matches (including Minting policies etc). We only want to pass if there's a match associated with the script output being spend with this condition.

> This is using as existing code (L8), so maybe have a different one?

I think the code is suitable because, unless there's some ledger bug, the absence of this particular `Spending` `ScriptPurpose` in redeemer map means the expected script output has not been spent.

Reply:
For the code, I meant that it is already used in `traceIfFalse "L8"`. Maybe change it to "L8a"?

Reply:
I've changed it now so there's only one `"L8"` in the code and all related failures return that.